Test Failure: PersistenceException and other FFDCs on testMultipleInstancesCompeteToRunManyLateTasks

Test Failure: com.ibm.ws.concurrent.persistent.fat.failover1serv.Failover1ServerTest2.testMultipleInstancesCompeteToRunManyLateTasks

The test case otherwise runs fine, but reports an error for FFDC messages that are logged when task executions don't run quickly enough and roll back. Given that the task executions later succeed when run either on the same or another instance, which is the point of failover, this test case should just be updated to expect the possibility of the FFDC messages.  This was partly already done under #10026 (XAException and a few others), however, a couple of additional FFDCs such as PersistenceException have been identified, which need to be added.

```
testMultipleInstancesCompeteToRunManyLateTasks:junit.framework.AssertionFailedError: 2019-12-05-11:52:34:618 [Unexpected FFDC reporting javax.transaction.xa.XAException was found (count = 5): /WebSphere/Misc/jazz_build/jbe_zrock162/jazz/buildsystem/buildengine/eclipse/build/dev/image/output/wlp/usr/servers/com.ibm.ws.concurrent.persistent.fat.failover1serv/logs/ffdc/ffdc_19.12.05_11.52.23.0.log
>------Start of DE processing------ = [12/5/19 11:52:23:978 GMT]
>Exception = javax.transaction.xa.XAException
>Source = com.ibm.ws.rsadapter.spi.WSRdbXaResourceImpl.end
>probeid = 438
>Stack Dump = javax.transaction.xa.XAException
>	at org.apache.derby.jdbc.EmbedXAResource.end(Unknown Source)
>	at com.ibm.ws.rsadapter.impl.WSRdbXaResourceImpl.end(WSRdbXaResourceImpl.java:417)
>	at com.ibm.ws.rsadapter.impl.WSRdbXaResourceImpl.end(WSRdbXaResourceImpl.java:337)
>	at com.ibm.ws.rsadapter.impl.WSRdbManagedConnectionImpl.cleanupTransactions(WSRdbManagedConnectionImpl.java:3138)
>	at com.ibm.ws.rsadapter.impl.WSRdbManagedConnectionImpl.cleanup(WSRdbManagedConnectionImpl.java:2751)
>	at com.ibm.ejs.j2c.MCWrapper.cleanup(MCWrapper.java:1541)
>	at com.ibm.ejs.j2c.PoolManager.release(PoolManager.java:922)
>	at com.ibm.ejs.j2c.MCWrapper.releaseToPoolManager(MCWrapper.java:2127)
>	at com.ibm.ejs.j2c.MCWrapper.abortMC(MCWrapper.java:3271)
>	at com.ibm.ejs.j2c.AbortableXATransactionWrapper.abort(AbortableXATransactionWrapper.java:49)
>	at com.ibm.tx.jta.impl.RegisteredResources.abort(RegisteredResources.java:2857)
>	at com.ibm.tx.jta.impl.TransactionImpl.abortTransactionParticipants(TransactionImpl.java:2874)
>	at com.ibm.tx.jta.embeddable.impl.EmbeddableTransactionImpl.timeoutTransaction(EmbeddableTransactionImpl.java:638)
>	at com.ibm.tx.jta.embeddable.impl.EmbeddableTimeoutManager$TimeoutInfo.alarm(EmbeddableTimeoutManager.java:165)
>	at com.ibm.tx.jta.util.alarm.AlarmManagerImpl$AlarmListenerWrapper.run(AlarmManagerImpl.java:83)
>	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522)
>	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
>	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:191)
>	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
>	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
>	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
>	at java.lang.Thread.run(Thread.java:812)
, Unexpected FFDC reporting java.lang.IllegalStateException was found (count = 1): /WebSphere/Misc/jazz_build/jbe_zrock162/jazz/buildsystem/buildengine/eclipse/build/dev/image/output/wlp/usr/servers/com.ibm.ws.concurrent.persistent.fat.failover1serv/logs/ffdc/ffdc_19.12.05_11.52.24.0.log
>------Start of DE processing------ = [12/5/19 11:52:24:404 GMT]
>Exception = java.lang.IllegalStateException
>Source = com.ibm.ejs.j2c.poolmanager.FreePool.cleanupAndDestroyMCWrapper
>probeid = 1140
>Stack Dump = java.lang.IllegalStateException: cleanup: illegal state exception. State = STATE_INACTIVE MCW = 96294c35
>	at com.ibm.ejs.j2c.MCWrapper.cleanup(MCWrapper.java:1506)
>	at com.ibm.ejs.j2c.FreePool.cleanupAndDestroyMCWrapper(FreePool.java:459)
>	at com.ibm.ejs.j2c.FreePool.getFreeConnection(FreePool.java:605)
>	at com.ibm.ejs.j2c.PoolManager.reserve(PoolManager.java:1329)
>	at com.ibm.ejs.j2c.ConnectionManager.allocateMCWrapper(ConnectionManager.java:581)
>	at com.ibm.ejs.j2c.ConnectionManager.allocateConnection(ConnectionManager.java:314)
>	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:138)
>	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:112)
>	at org.eclipse.persistence.sessions.JNDIConnector.connect(JNDIConnector.java:135)
>	at org.eclipse.persistence.sessions.DatasourceLogin.connectToDatasource(DatasourceLogin.java:162)
>	at org.eclipse.persistence.internal.databaseaccess.DatasourceAccessor.connectInternal(DatasourceAccessor.java:346)
>	at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.connectInternal(DatabaseAccessor.java:309)
>	at org.eclipse.persistence.internal.databaseaccess.DatasourceAccessor.reconnect(DatasourceAccessor.java:581)
>	at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.reconnect(DatabaseAccessor.java:1634)
>	at org.eclipse.persistence.internal.databaseaccess.DatasourceAccessor.incrementCallCount(DatasourceAccessor.java:321)
>	at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.basicExecuteCall(DatabaseAccessor.java:615)
>	at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.executeCall(DatabaseAccessor.java:560)
>	at org.eclipse.persistence.internal.sessions.AbstractSession.basicExecuteCall(AbstractSession.java:2064)
>	at org.eclipse.persistence.sessions.server.ClientSession.executeCall(ClientSession.java:310)
>	at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.executeCall(DatasourceCallQueryMechanism.java:262)
>	at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.executeCall(DatasourceCallQueryMechanism.java:248)
>	at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.executeNoSelectCall(DatasourceCallQueryMechanism.java:291)
>	at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.updateAll(DatasourceCallQueryMechanism.java:873)
>	at org.eclipse.persistence.queries.UpdateAllQuery.executeDatabaseQuery(UpdateAllQuery.java:154)
>	at org.eclipse.persistence.queries.DatabaseQuery.execute(DatabaseQuery.java:908)
>	at org.eclipse.persistence.queries.DatabaseQuery.executeInUnitOfWork(DatabaseQuery.java:807)
>	at org.eclipse.persistence.queries.ModifyAllQuery.executeInUnitOfWork(ModifyAllQuery.java:148)
>	at org.eclipse.persistence.internal.sessions.UnitOfWorkImpl.internalExecuteQuery(UnitOfWorkImpl.java:2900)
>	at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1866)
>	at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1848)
>	at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1813)
>	at org.eclipse.persistence.internal.jpa.QueryImpl.executeUpdate(QueryImpl.java:298)
>	at com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore.incrementFailureCount(DatabaseTaskStore.java:1186)
>	at com.ibm.ws.concurrent.persistent.internal.InvokerTask.processRetryableTaskFailure(InvokerTask.java:165)
>	at com.ibm.ws.concurrent.persistent.internal.InvokerTask.run(InvokerTask.java:607)
>	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522)
>	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
>	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
>	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
>	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
>	at java.lang.Thread.run(Thread.java:812)
, Unexpected FFDC reporting javax.persistence.PersistenceException was found (count = 5): /WebSphere/Misc/jazz_build/jbe_zrock162/jazz/buildsystem/buildengine/eclipse/build/dev/image/output/wlp/usr/servers/com.ibm.ws.concurrent.persistent.fat.failover1serv/logs/ffdc/ffdc_19.12.05_11.52.23.2.log
>------Start of DE processing------ = [12/5/19 11:52:23:973 GMT]
>Exception = javax.persistence.PersistenceException
>Source = com.ibm.ws.concurrent.persistent.internal.InvokerTask
>probeid = 581
>Stack Dump = javax.persistence.PersistenceException: 
>Exception Description: Error binding to externally managed transaction
>Internal Exception: javax.transaction.RollbackException: Transaction rolled back
>	at org.eclipse.persistence.internal.jpa.transaction.JTATransactionWrapper.registerIfRequired(JTATransactionWrapper.java:145)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.setJTATransactionWrapper(EntityManagerImpl.java:2168)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.detectTransactionWrapper(EntityManagerImpl.java:893)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.initialize(EntityManagerImpl.java:416)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.<init>(EntityManagerImpl.java:409)
>	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.createEntityManagerImpl(EntityManagerFactoryDelegate.java:324)
>	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManagerImpl(EntityManagerFactoryImpl.java:337)
>	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManager(EntityManagerFactoryImpl.java:303)
>	at com.ibm.wsspi.persistence.internal.PersistenceServiceUnitImpl.createEntityManager(PersistenceServiceUnitImpl.java:96)
>	at com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore.persist(DatabaseTaskStore.java:1375)
>	at com.ibm.ws.concurrent.persistent.internal.InvokerTask.run(InvokerTask.java:548)
>	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522)
>	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
>	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
>	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
>	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
>	at java.lang.Thread.run(Thread.java:812)
>Caused by: javax.transaction.RollbackException: Transaction rolled back
>	at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2285)
>	at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2265)
>	at org.eclipse.persistence.internal.jpa.transaction.JTATransactionWrapper.registerIfRequired(JTATransactionWrapper.java:136)
>	... 16 more
, Unexpected FFDC reporting javax.transaction.RollbackException was found (count = 5): /WebSphere/Misc/jazz_build/jbe_zrock162/jazz/buildsystem/buildengine/eclipse/build/dev/image/output/wlp/usr/servers/com.ibm.ws.concurrent.persistent.fat.failover1serv/logs/ffdc/ffdc_19.12.05_11.52.23.1.log
>------Start of DE processing------ = [12/5/19 11:52:23:985 GMT]
>Exception = javax.transaction.RollbackException
>Source = com.ibm.tx.jta.TransactionImpl.registerSynchronization
>probeid = 2289
>Stack Dump = javax.transaction.RollbackException: Transaction rolled back
>	at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2285)
>	at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2265)
>	at org.eclipse.persistence.internal.jpa.transaction.JTATransactionWrapper.registerIfRequired(JTATransactionWrapper.java:136)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.setJTATransactionWrapper(EntityManagerImpl.java:2168)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.detectTransactionWrapper(EntityManagerImpl.java:893)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.initialize(EntityManagerImpl.java:416)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.<init>(EntityManagerImpl.java:409)
>	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.createEntityManagerImpl(EntityManagerFactoryDelegate.java:324)
>	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManagerImpl(EntityManagerFactoryImpl.java:337)
>	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManager(EntityManagerFactoryImpl.java:303)
>	at com.ibm.wsspi.persistence.internal.PersistenceServiceUnitImpl.createEntityManager(PersistenceServiceUnitImpl.java:96)
>	at com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore.persist(DatabaseTaskStore.java:1375)
>	at com.ibm.ws.concurrent.persistent.internal.InvokerTask.run(InvokerTask.java:548)
>	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522)
>	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
>	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
>	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
>	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
>	at java.lang.Thread.run(Thread.java:812)
]
	at componenttest.custom.junit.runner.FATRunner.blowup(FATRunner.java:296)
	at componenttest.custom.junit.runner.FATRunner.access$600(FATRunner.java:64)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:246)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:30)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:314)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:167)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:24)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:39)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:520)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:1060)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:911)
```

--- a/dev/com.ibm.ws.concurrent.persistent_fat_failover1serv/fat/src/com/ibm/ws/concurrent/persistent/fat/failover1serv/Failover1ServerTest2.java
+++ b/dev/com.ibm.ws.concurrent.persistent_fat_failover1serv/fat/src/com/ibm/ws/concurrent/persistent/fat/failover1serv/Failover1ServerTest2.java
@@ -86,6 +86,8 @@ public class Failover1ServerTest2 extends FATServletClient {
     @AllowedFFDC(value = {
             "javax.transaction.RollbackException", // transaction rolled back due to timeout
             "javax.transaction.xa.XAException", // rollback/abort path
+            "javax.persistence.PersistenceException", // caused by RollbackException
+            "javax.persistence.ResourceException", // connection error event on retry
             "java.lang.IllegalStateException" // for EclipseLink retry after connection has been aborted due to rollback
     })
     @Test

